lib/kdeproject_component.rb

```diff
@@ -42,8 +44,13 @@ def plasma
 
     private
 
-    def to_names(projects)
-      projects.collect { |project| project.split('/')[-1] }
+    def to_names(project_list)
+      project_list.collect! { |project| project.split('/')[-1] }
+      @projects_to_jobs.each do |project_name, job_name|
```

Comment:
This entire block is better expressed as 

```
project_list.collect { |x| @projects_to_jobs.fetch(x, x) }
```
which re-collects the array, by running each value through the hash with both the value as key AND default value. i.e. `fetch('foobar', 'foobar')` = if 'foobar' is in the hash it will return the value it is mapped to, otherwise it will return 'foobar'